nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/adls/PutADLSFile.java

```diff
@@ -0,0 +1,282 @@
+            .name("adls-put-conflict-resolution-strategy")
+            .displayName("Conflict Resolution Strategy")
+            .description("Indicates what should happen when a file " +
+                    "with the same name already exists in the output directory")
+            .required(true)
+            .defaultValue(FAIL_RESOLUTION_AV.getValue())
+            .allowableValues(REPLACE_RESOLUTION_AV, FAIL_RESOLUTION_AV, APPEND_RESOLUTION_AV)
+            .build();
+
+    protected static final PropertyDescriptor DIRECTORY = new PropertyDescriptor.Builder()
+            .name("adls-put-directory")
+            .displayName("Directory")
+            .description("The directory where files will be written")
+            .required(true)
+            .addValidator(StandardValidators.ATTRIBUTE_EXPRESSION_LANGUAGE_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    protected static final PropertyDescriptor UMASK = new PropertyDescriptor.Builder()
+            .name("adls-put-permissions-umask")
+            .displayName("Permissions umask")
+            .description(
+                    "A umask represented as an octal number which determines the permissions of files written to ADLS.")
+            .addValidator(ADLSValidators.UMASK_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor ACL = new PropertyDescriptor.Builder()
+            .name("adls-put-access-control-list")
+            .displayName("Access Control List")
+            .description(
+                    "Comma separated list of ACL entry. An ACL entry consists of a scope (access or default)" +
+                            " the type of the ACL (user, group, other or mask), the name of the user or group" +
+                            " associated with this ACL (can be blank to specify the default permissions for" +
+                            " users and groups, and must be blank for mask entries), and the action permitted" +
+                            " by this ACL entry. eg: \"default:user:bob:r-x,default:user:adam:rw\"")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    @Override
+    protected void init(ProcessorInitializationContext context) {
+        super.init(context);
+
+        super.descriptors.add(CONFLICT_RESOLUTION);
+        super.descriptors.add(DIRECTORY);
+        super.descriptors.add(UMASK);
+        super.descriptors.add(ACL);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final StopWatch dataRateStopWatch = new StopWatch();
+        final StopWatch completeProcessStopWatch = new StopWatch(true);
+
+        ADLStoreClient adlsClient = getAdlStoreClient();
+        //safe check
```

Comment:
This check for `null` is duplicated in a few places. Is it possible to move the check to `getAdlStoreClient()`, and if it is going to return `null`, then create the client (`createADLSClient()`) in order to get rid of the checks?

Reply:
@jzonthemtn  Yes, it makes sense to push the lazy initialization to Getter, did that. Thanks for review.